### PR TITLE
[WIP] Vegetarian vegan palm

### PR DESCRIPTION
**Pull request by teolemon**

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -983,4 +987,27 @@
     <string name="cant_modify_if_refreshing">Please wait until the upload finishes</string>
     <string name="closeZoom">Click to close zoom</string>
     <string name="imageFullscreen">Zoom</string>
+
+    <!-- Used for product ingredients analysis tags -->
+    <string name="close_button">Close</string>
+    <string name="tip_message">TIP: %1$s </string>
+    <string name="ingredient_analysis_tip">Click on the icons bellow to learn more about the product and to select what information you are interested in.</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="ingredient_analysis_tip">Click on the icons below to learn more about the product and to select what information you are interested in.</string>
```

**Comment by teolemon** [comment]
Issue: it's impossible to scroll the product view on the vegan/veg/palm branch

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsWithTagDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_sad);
+                            break;
+                        case "yes":
+                            titleResId = R.string.vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_happy);
+                            messageStr = getString(R.string.this_product_is, getString(titleResId).toLowerCase());
+                            break;
+                        default:
+                            titleResId = R.string.unknown_vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_unknown);
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsWithTagDialogFragment.java]: I would extract those colors as an array or something, and give them a name that's more generic
            R.color.monkey_uncertain
            R.color.monkey_unknown
            R.color.monkey_sad
            R.color.monkey_happy

            R.color.detection_uncertain
            R.color.detection_unknown
            R.color.detection_green
            R.color.detection_red

**Comment by teolemon** [comment]
``` 
2019-10-13 15:09:09.404 6294-6294/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.openfoodfacts.scanner, PID: 6294
    j.b.a.h.a
        at j.b.a.h.b.b(AsyncOperation.java:5)
        at openfoodfacts.github.scrachx.openfood.fragments.m1.a(PreferencesFragment.java:89)
        at openfoodfacts.github.scrachx.openfood.fragments.z0.a(Unknown Source:2)
        at j.b.a.h.c.handleMessage(AsyncOperationExecutor.java:2)
        at android.os.Handler.dispatchMessage(Handler.java:103)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: android.database.sqlite.SQLiteException: no such table: ANALYSIS_TAG_CONFIG (code 1 SQLITE_ERROR): , while compiling: SELECT T."_id",T."ANALYSIS_TAG",T."TYPE",T."ICON",T."COLOR" FROM "ANALYSIS_TAG_CONFIG" T  WHERE 1 GROUP BY type ORDER BY T.'TYPE' COLLATE NOCASE ASC
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:986)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:593)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1443)
        at android.database.sqlite.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1382)
        at j.b.a.i.f.a(StandardDatabase.java:1)
        at j.b.a.l.e.c(Query.java:2)
        at j.b.a.h.c.b(AsyncOperationExecutor.java:11)
        at j.b.a.h.c.c(AsyncOperationExecutor.java:1)
        at j.b.a.h.c.run(AsyncOperationExecutor.java:14)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

File: app/build.gradle [hunk 3]
```diff
@@ -25,8 +25,8 @@ android {
         applicationId "openfoodfacts.github.scrachx.openfood"
         minSdkVersion 16
         targetSdkVersion 28
-        versionCode 322
-        versionName "3.2.2"
+        versionCode 323
+        versionName "3.2.1"
```
Comment (teolemon) [inline app/build.gradle]:
```suggestion
        versionName "3.2.3"
```

**Comment by teolemon** [comment]
I relaunched the build. It's green now.

**Comment by teolemon** [comment]
@ripetrescu had added a migration already, weird. But ok to bump schema numbers.

**Comment by ripetrescu** [comment]
It's weird. I already bumped the schema to 14 and the last check in migration is 13. Also, there are only new tables, so it should be no conflict with previous one:
...
            case 13: {
                AnalysisTagNameDao.createTable(db, true);
                AnalysisTagDao.createTable(db, true);
                AnalysisTagConfigDao.createTable(db, true);
                break;
...

**Comment by deniger** [comment]
I suppose it's certainly due to another PR i tested ( ingredient) containing migrations too.
To be sure, I prefer to use the version 15 :)

**Comment by teolemon** [comment]
ok, let's bump.

**Comment by deniger** [comment]
tested on android 4 -> ok

**Comment by deniger** [comment]
gg @ripetrescu  👍

**Comment by deniger** [comment]
@teolemon can be merged...

**Comment by teolemon** [comment]
I'll leave you the honor to merge @ripetrescu 👌

**Comment by teolemon** [comment]
Very fine work @ripetrescu and thank you very much @deniger for the corrections

**Comment by ripetrescu** [comment]
Done. Thank you guys! :)

**Pull request merged**